src/server.rs

```diff
@@ -564,18 +619,34 @@ impl ServerState {
                             return Err(Box::new(io::Error::new(ErrorKind::PermissionDenied, "invalid cookie")));
                         }
                     };
+
+/*                    {
+                        let player_next_sequence_num: u64 = self.get_player(player_id).next_resp_seq;
+                        println!("{} <= {}", sequence, player_next_sequence_num - 1);
+                        if sequence <= (player_next_sequence_num - 1) && action != RequestAction::KeepAlive {
+                            self.silence_error = true;
+                            return Err(Box::new(io::Error::new(ErrorKind::AlreadyExists, "")));
+                        }
+                    }
+*/
                     match action {
                         RequestAction::Connect{..} => unreachable!(),
                         _ => {
-                            let response = self.handle_request_action(player_id, action);
-                            Ok(Some(response))
+                            let response = self.handle_request_action(player_id, action.clone());
+
+                            if action != RequestAction::KeepAlive {
```

Comment:
Wait, don't we need to respond to a `KeepAlive` so that the client doesn't decide the connection is dead if nothing's going on?